src/common/transformations/src/transformations/transpose_sinking/ts_utils.cpp

```diff
@@ -68,9 +68,20 @@ Output<Node> ChangeAxes(const Output<Node>& indices,
     return ChangeAxes(indices, data, axis);
 }
 
-TransposeInputsInfo GetFirstTransposeInput(const NodePtr& node,
-                                           bool const_transpose_order,
-                                           const std::vector<size_t>& indices) {
+bool if_transpose_sinkable_default(const std::shared_ptr<ov::op::v1::Transpose>& transpose,
+                                   const std::shared_ptr<ov::op::v0::Constant>& transpose_order) {
+    if (!transpose || !transpose_order)
+        return false;
+    const auto partial_shape_rank = transpose->get_input_partial_shape(0).rank();
+    const auto order = transpose_order->get_axis_vector_val();
+    return (partial_shape_rank.is_static() || order.empty());
```

Comment:
Should it be 
`!(partial_shape_rank.is_dynamic() && order.empty());` ?

In your implementation it returns "true" if order is empty, but rank is dynamic